DOCS-1618: Update README.md (#3)
viam-labs/facial-detection: README.md

# Facial Detector modular resource

This module implements the [vision API](https://python.viam.dev/autoapi/viam/services/vision/client/index.html) in the `facial-detector` model and leverages the [DeepFace](https://github.com/serengil/deepface) library to perform facial detections.

## Requirements

This module is compatible with the following platforms:

- MacOS (darwin)
- Linux

Run the following command to get your operating system name, and then proceed to the instructions corresponding to your operating system.

```bash
uname -s
```

On MacOS:

`run.sh` will automatically install the following system dependencies if not already set up on the machine:

- `viam-sdk  >= 0.5.1`
- `deepface`
- `numpy`
- `pillow`
- `tensorflow==2.14.0rc1`
- `ml_dtypes==0.2.0`

On Linux:

``` bash
sudo apt update && sudo apt upgrade -y
sudo apt-get update && sudo apt-get install ffmpeg libsm6 libxext6  -y
```

## Configure your facial detector

> [!NOTE]

> > Before configuring your facial-detctor, you must [create a machine](https://docs.viam.com/manage/fleet/machines/#add-a-new-machine).

Author: ```suggestion
> Before configuring your facial-detector, you must [create a machine](https://docs.viam.com/manage/fleet/machines/#add-a-new-machine).
```

Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com/).
Click on the **Services** subtab and click **Create service**.
Select the `vision` type, then select the `detector:facial-detector` model. 
Click **Add module**, then enter a name for your facial detector and click **Create**.

> On the new component panel, copy and paste the following attribute template into your vision service's **Attributes** box:

Author: ```suggestion
On the new service panel, copy and paste the following attribute template into your vision service's **Attributes** box:
```

## Requirements

`run.sh` will automatically install the following system dependencies if not already set up on the machine:

- `viam-sdk  >= 0.5.1`
- `deepface`
- `numpy`
- `pillow`
- `tensorflow==2.14.0rc1`
- `ml_dtypes==0.2.0`

On Linux machines, `run.sh` will also install the following:

- `ffmpeg`
- `libsm6`
- `libxext6`

## Configure your facial detector

> [!NOTE]
> Before configuring your facial-detector, you must [create a machine](https://docs.viam.com/manage/fleet/machines/#add-a-new-machine).

Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com/).
Click on the **Services** subtab and click **Create service**.
Select the `vision` type, then select the `detector:facial-detector` model. 
Click **Add module**, then enter a name for your facial detector and click **Create**.

On the new component panel, copy and paste the following attribute template into your vision service's **Attributes** box:

> ```json 

Reviewer: Remove unneeded whitespace
```suggestion
```json
```

}
```

> [!NOTE]
> For more information, see [Configure a Machine](https://docs.viam.com/manage/configuration/).

### Attributes 

The following attributes are available for `vision:facial-detector:ssd` facial-detectors:

| Name   | Type | Inclusion | Description| Supported |
|--------|-------|----------|------------|-----------|
| `detection_framework` | enum | Optional| The detection framework to use for facial detection.<br> Default: `ssd`, for a good balance of speed and accuracy. | `"opencv"`, `"retinaface"`, `"mtcnn"`, `"ssd"`, `"dlib"`, `"mediapipe"`, `"yolov8"` |
| `recognition_model` | enum | Optional| The model to use for facial recognition.<br> Default: `ArcFace`, for a good balance of speed and accuracy.| `"VGG-Face"`, `"Facenet"`, `"Facenet512"`, `"OpenFace"`, `"DeepFace"`, `"DeepID"`, `"ArcFace"`, `"Dlib"`, `"SFace"` |                     |
| `face_labels` | object  | Optional | An object map of `key:label, value:path` for matching against reference face images. | - |
| `verify_threshold` | number  | Optional | If `disable_verify` is `false` and `face_labels` are set, this threshold must be met or exceeded for a verification match.<br> Default: `0.8` | - |

> | `disable_detect` | bool | Optional| If `false`, any faces detected but not verified as matching a label will be labeled as "face".<br> Default: `false` | - |
> | `disable_verify` | bool | Optional | If `false` and `face_labels` are set, will attempt to verify any faces detected. If you only want verified faces, set this and `disable_detect` to `true`.<br> Default: `false` | - |

Reviewer: Could otherwise sound like false is set
```suggestion
| `disable_detect` | bool | Optional| If `false`, any faces detected but not verified as matching a label will be labeled as "face".<br> Default: `false` | - |
| `disable_verify` | bool | Optional | If set to `false` and `face_labels` are set, will attempt to verify any faces detected. If you only want verified faces, set this and `disable_detect` to `true`.<br> Default: `false` | - |
```

| `recognition_model` | enum | Optional| The model to use for facial recognition.<br> Default: `ArcFace`, for a good balance of speed and accuracy.| `"VGG-Face"`, `"Facenet"`, `"Facenet512"`, `"OpenFace"`, `"DeepFace"`, `"DeepID"`, `"ArcFace"`, `"Dlib"`, `"SFace"` |                     |
| `face_labels` | object  | Optional | An object map of `key:label, value:path` for matching against reference face images. | - |
| `verify_threshold` | number  | Optional | If `disable_verify` is `false` and `face_labels` are set, this threshold must be met or exceeded for a verification match.<br> Default: `0.8` | - |
| `disable_detect` | bool | Optional| If `false`, any faces detected but not verified as matching a label will be labeled as "face".<br> Default: `false` | - |
| `disable_verify` | bool | Optional | If `false` and `face_labels` are set, will attempt to verify any faces detected. If you only want verified faces, set this and `disable_detect` to `true`.<br> Default: `false` | - |

> *object*

Reviewer: I think this maybe has gone missing as an attribute?

Author: I don't think I understand what you mean by this... are you saying the `face_labels` attribute may have been deprecated or possibly that there is missing info in the **Attributes** table I wrote?

> The `facial-detector` resource provides the following methods from [Viam's vision API](https://python.viam.dev/autoapi/viam/services/vision/client/index.html).

Reviewer: ```suggestion
The `facial-detector` resource provides the following methods from [Viam's vision API](https://python.viam.dev/autoapi/viam/services/vision/client/index.html):
```

> [Create a Facial Verification System](https://docs.viam.com/tutorials/projects/verification-system/#configure-a-verification-system) using the facial detector you configured.

Reviewer: Is there anything else we can suggest to people just to check that the module is working? I think we usually suggest something they can confirm on the control tab.
And then I'd suggest making this sound like an option the person has. They can use this module for other purposes after all:

```suggestion
Check out the [Create a Facial Verification System](https://docs.viam.com/tutorials/projects/verification-system/#configure-a-verification-system) tutorial that uses this module.
```
